common/lib/xmodule/xmodule/modulestore/tests/test_mixed_modulestore.py

```diff
@@ -2091,3 +2093,33 @@ def test_import_edit_import(self, default):
                 with self.store.branch_setting(ModuleStoreEnum.Branch.published_only, course_id):
                     published_vertical = self.store.get_item(vertical_loc)
                 self.assertEqual(draft_vertical.display_name, published_vertical.display_name)
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_course_publish_signal_firing(self, default):
+        with MongoContentstoreBuilder().build() as contentstore:
+            self.store = MixedModuleStore(
+                contentstore=contentstore,
+                create_modulestore_instance=create_modulestore_instance,
+                mappings={},
+                signal_handler=SignalHandler(MixedModuleStore),
+                **self.OPTIONS
+            )
+            self.addCleanup(self.store.close_all_connections)
+
+            with self.store.default_store(default):
+                self.assertIsNotNone(self.store.thread_cache.default_store.signal_handler)
+
+                # Course creation should fire the signal
+                with mock_signal_receiver(SignalHandler.course_published) as receiver:
+                    self.assertEqual(receiver.call_count, 0)
+                    course = self.store.create_course('org_x', 'course_y', 'run_z', self.user_id)
+                    self.assertEqual(receiver.call_count, 1)
+
+                    course_key = course.id
+                    receiver.reset_mock()
+                    self.store.create_item(self.user_id, course_key, 'problem')
+                    self.assertEqual(receiver.call_count, 1)
+
+                    receiver.reset_mock()
+                    self.store.publish(course.location, self.user_id)
+                    self.assertEqual(receiver.call_count, 1)
```

Comment:
@ormsbee: Please confirm that the following is the desired behavior for this PR.

*Additional tests:*
1. import course -> call_count should be 1

2. update a module that is not in `DIRECT_ONLY_CATEGORIES` (e.g., update a block of type 'problem'):
  update_item -> call_count should be *0*  @ormsbee: please confirm
  publish_item -> call_count should be 1

3. update a module that is in `DIRECT_ONLY_CATEGORIES` (e.g., update a block of type 'course_info'):
  update_item -> call_count should 1